addons/survey/views/survey_templates.xml

```diff
@@ -597,5 +588,31 @@
             </div>
         </t>
     </template>
+
+    <template id="survey_navigation" name="Survey: Navigation">
+        <t t-set="can_go_back" t-value="survey and answer and answer.state == 'in_progress' and survey.users_can_go_back" />
+        <!-- Can go back in 'page_per_section' if:
+            - Back button is allowed in configuration
+            - This is not the first section
+        -->
+        <t t-set="page_per_section_back_button" t-value="can_go_back and survey.questions_layout == 'page_per_section' and page != survey.page_ids[0]" />
+        <!-- Can go back in 'page_per_question' if:
+            - Back button is allowed in configuration
+            - This is not a session answer (doesn't make sense to go back in session context)
+            - This is not the first question
+            - The survey doesn't have pages OR this is not the first page of the survey
+                (pages are displayed in 'page_per_question' layout when they have a description, see PR#44271)
+        -->
+        <t t-set="page_per_question_back_button"
```

Comment:
[C6] At one point it becomes easier to prepare values in controllers than trying to understand complex variable value in template. I would move it to the render, especially that it is already rendered server-side in a specific render call.